Sources/Signal.swift

```diff
@@ -1578,6 +1655,20 @@ private struct ThrottleState<Value> {
 	var pendingValue: Value? = nil
 }
 
+private enum ThrottleWhileState<Value> {
+	case resumed
+	case throttled(Value?)
+	case terminated
+
+	var isTerminated: Bool {
+		if case .terminated = self {
+			return true
+		} else {
+			return false
+		}
+	}
```

Comment:
Stylistically, this could also become a switch statement if you prefer, which would add exhaustively checking.